plugins/package-managers/maven/src/main/kotlin/utils/P2ArtifactTracker.kt

```diff
@@ -149,9 +141,8 @@ class P2ArtifactTracker(
      */
     private fun findHash(artifact: Artifact): Hash =
         repositoryHelper.p2Properties(artifact)?.let { properties ->
-            hashAlgorithms.firstOrNull { "$CHECKSUM_PROPERTY_PREFIX${it.first}" in properties }
-                ?.let { (size, algorithm) ->
-                    Hash(properties.getValue("${CHECKSUM_PROPERTY_PREFIX}$size"), algorithm)
-                }
+            properties.map { (key, value) ->
+                Hash(value, key.removePrefix(CHECKSUM_PROPERTY_PREFIX))
```

Comment:
Wouldn't this fail for properties not referring to checksums? How does `Hash`behave if an unexpected algorithm name is passed in?

Reply:
I see. Would it then be better to filter for properties starting with `CHECKSUM_PROPERTY_PREFIX` to prevent the unnecessary construction of meaningless `Hash` objects?